### PR TITLE
Ftr prefered language

### PR DESCRIPTION
**Pull request by ansmonjol**
## Context

For legal reason, some companies had to generate invoices in their own languages.

## Description

This PR let users set preferedLocale on both organization and customer level (customer one being an override of the organization value if set)

## Side note

This PR introduces 2 TODO as comments. Those have been logged in our internal tool and will be tackled later

### CODE REVIEW

File: src/components/customers/CustomerSettings.tsx [hunk 1]
```diff
@@ -281,6 +300,90 @@ export const CustomerSettings = ({ customerId }: CustomerSettingsProps) => {
...
+        {loading ? (
+          <>
+            <Skeleton variant="text" width={320} height={12} marginBottom={theme.spacing(4)} />
+            <Skeleton variant="text" width={160} height={12} />
+          </>
+        ) : (
+          <>
+            <Typography variant="body" color="grey700">
+              {typeof customer?.billingConfiguration?.documentLocale === 'string'
+                ? // @ts-ignore
```
Comment (MorganeLecurieux) [inline src/components/customers/CustomerSettings.tsx:367]: Why do you need to `ts-ignore` here ?

File: src/components/customers/CustomerSettings.tsx [hunk 2]
```diff
@@ -281,6 +300,90 @@ export const CustomerSettings = ({ customerId }: CustomerSettingsProps) => {
...
+                  DocumentLocales[customer.billingConfiguration.documentLocale]
+                : translate('text_63ea0f84f400488553caa773', {
+                    // @ts-ignore
```
Comment (MorganeLecurieux) [inline src/components/customers/CustomerSettings.tsx:370]: Why do you need to `ts-ignore` here ?

File: src/components/customers/EditCustomerDocumentLocaleDialog.tsx [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+
+interface EditCustomerDocumentLocaleDialogProps {
+  customer: EditCustomerDocumentLocaleFragment
+}
+
+const documentLocalesData: { value: string; label: string }[] = Object.keys(DocumentLocales).map(
+  (localeKey) => {
+    return {
+      value: localeKey,
+      // @ts-ignore
```
Comment (MorganeLecurieux) [inline src/components/customers/EditCustomerDocumentLocaleDialog.tsx:50]: Why ts-ignore ?

File: src/components/customers/EditCustomerDocumentLocaleDialog.tsx [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+    }),
+    enableReinitialize: true,
+    validateOnMount: true,
+    onSubmit: async (values) => {
+      await updateDocumentLocale({
+        variables: {
+          input: {
+            id: customer.id,
+            ...values,
+            // TODO: Required by API, will fix those non-sense requirements later
```
Comment (MorganeLecurieux) [inline src/components/customers/EditCustomerDocumentLocaleDialog.tsx:93]: What's the non-sense here ?

File: src/components/customers/EditCustomerDocumentLocaleDialog.tsx [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        formikProps.resetForm()
+        formikProps.validateForm()
+      }}
+      actions={({ closeDialog }) => (
+        <>
+          <Button
+            variant="quaternary"
+            onClick={() => {
+              closeDialog()
+              formikProps.resetForm()
```
Comment (MorganeLecurieux) [inline src/components/customers/EditCustomerDocumentLocaleDialog.tsx:119]: No need to also `validateForm` here ?

File: src/components/customers/EditCustomerDocumentLocaleDialog.tsx [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+          >
+            {translate('text_63ea0f84f400488553caa6a5')}
+          </Button>
+          <Button
+            variant="primary"
+            disabled={!formikProps.isValid || !formikProps.dirty}
+            onClick={async () => {
+              await formikProps.submitForm()
+              closeDialog()
+              formikProps.resetForm()
```
Comment (MorganeLecurieux) [inline src/components/customers/EditCustomerDocumentLocaleDialog.tsx:130]: No need `validateForm` here as well ?

File: src/components/settings/EditOrganizationDocumentLocaleDialog.tsx [hunk 7]
```diff
@@ -0,0 +1,142 @@
...
+
+interface EditOrganizationDocumentLocaleDialogProps {
+  documentLocale: string
+}
+
+const documentLocalesData: { value: string; label: string }[] = Object.keys(DocumentLocales).map(
+  (localeKey) => {
+    return {
+      value: localeKey,
+      // @ts-ignore
```
Comment (MorganeLecurieux) [inline src/components/settings/EditOrganizationDocumentLocaleDialog.tsx:40]: Why the ignore ?

File: src/components/customers/EditCustomerDocumentLocaleDialog.tsx (hunk 4 above)
Comment (ansmonjol) [reply src/components/customers/EditCustomerDocumentLocaleDialog.tsx:93]: There is no sense to me to send the customer name and externalD in order to update a different attribute.

Only ID should be required imo.

Those are set as required today for some reason and I opened an improvement ticket to look into it and take more time on it later

File: src/pages/settings/InvoiceSettings.tsx [hunk 8]
```diff
@@ -148,6 +156,41 @@ const InvoiceSettings = () => {
...
+      <InfoBlock>
+        {loading ? (
+          <>
+            <Skeleton variant="text" width={320} height={12} marginBottom={theme.spacing(4)} />
+            <Skeleton variant="text" width={160} height={12} />
+          </>
+        ) : (
+          <>
+            <Typography variant="body" color="grey700">
+              {/* @ts-ignore */}
```
Comment (MorganeLecurieux) [inline src/pages/settings/InvoiceSettings.tsx:181]: Why @ts-ignore ?

File: src/pages/settings/InvoiceSettings.tsx [hunk 9]
```diff
@@ -148,6 +156,41 @@ const InvoiceSettings = () => {
...
+              {DocumentLocales[documentLocale]}
+            </Typography>
+            <Typography variant="caption" color="grey600">
+              {translate('text_63e51ef4985f0ebd75c212ff', {
+                // @ts-ignore
```
Comment (MorganeLecurieux) [inline src/pages/settings/InvoiceSettings.tsx:186]: Why @ts-ignore ?

File: src/components/customers/EditCustomerDocumentLocaleDialog.tsx (hunk 5 above)
Comment (ansmonjol) [reply src/components/customers/EditCustomerDocumentLocaleDialog.tsx:119]: Form cannot be submit without value selected, so no error will be displayed bellow the input.

So there is no need to validate the form again !

File: src/components/customers/CustomerSettings.tsx (hunk 1 above)
Comment (ansmonjol) [reply src/components/customers/CustomerSettings.tsx]: documentLocales is now typed, all ignore have been removed in last commit

**Pull request merged**